ext/dom/element.c

```diff
@@ -1111,14 +1208,28 @@ PHP_METHOD(DOMElement, getElementsByTagNameNS)
 
 	DOM_GET_THIS_INTERN(intern);
 
-	php_dom_create_iterator(return_value, DOM_NODELIST);
+	if (modern) {
+		php_dom_create_iterator(return_value, DOM_HTMLCOLLECTION, true);
+	} else {
+		php_dom_create_iterator(return_value, DOM_NODELIST, false);
+	}
 	namednode = Z_DOMOBJ_P(return_value);
 	dom_namednode_iter(intern, 0, namednode, NULL, name, name_len, uri ? uri : "", uri_len);
```

Comment:
Can't `uri_len` here be uninitialized if `uri` is `NULL`?

TBF looking at the code, it seems like `uri` should always be initialized, but `name_len` might not be?

Reply:
> Can't uri_len here be uninitialized if uri is NULL?

No. https://github.com/php/php-src/blob/d9549d2ee215cb04aa5d2e3195c608d581fb272c/Zend/zend_API.h#L2296 When NULL the length becomes 0.

> but name_len might not be?

Why not? The name argument is not optional.

Reply:
Smh I interpreted the `!` as a `|`